users/views.py

```diff
@@ -1,8 +1,33 @@
 from users.models import CustomUser
-from users.serializers import UserSerializer
+from users.serializers import UserDetailSerializer, UserListSerializer, UserCreateSerializer, UserUpdateSerializer, UserSerializer
 from rest_framework import viewsets
+from rest_framework import permissions
+from users.permissions import IsAdminUser, IsOwnerOrAdmin
+
+
+class UsersViewSet(viewsets.ModelViewSet):
+    queryset = CustomUser.objects.all()
+    permission_classes = (IsAdminUser,)
+
+    def get_serializer_class(self):
+        if self.action == 'list':
```

Comment:
Would be fine if there is something like enum for that. `self.action == Action.LIST.value`. This is not critical, but we should do it after demo.

Reply:
Done